dynawo/sources/Models/Modelica/Dynawo/Electrical/Injectors/InjectorIDQ.mo

```diff
@@ -0,0 +1,81 @@
+  Blocks.Interfaces.RealInput iqPu (start = Iq0Pu) "Injected q-axis current (pu base SNom)" annotation(
+  Placement(visible = true, transformation(origin = {-110, -60}, extent = {{-20, -20}, {20, 20}}, rotation = 0), iconTransformation(origin = {-97, -1}, extent = {{-9, -9}, {9, 9}}, rotation = 0)));
+  
+  // Outputs:
+  Blocks.Interfaces.RealOutput UPu (start = U0Pu) "Magnitude voltage at inverter terminal (pu base UNom)" annotation(
+    Placement(visible = true, transformation(origin = {110, 89}, extent = {{-17, -17}, {17, 17}}, rotation = 0), iconTransformation(origin =  {102, 80}, extent = {{-8, -8}, {8, 8}}, rotation = 0)));
+  Blocks.Interfaces.RealOutput QInjPu (start = -Q0Pu) "injected reactive power (pu base SNom)" annotation(
+    Placement(visible = true, transformation(origin = {110, 21}, extent = {{-17, -17}, {17, 17}}, rotation = 0), iconTransformation(origin =  {102, 20}, extent = {{-8, -8}, {8, 8}}, rotation = 0)));
+  Blocks.Interfaces.RealOutput PInjPu (start = -P0Pu) "injected active power (pu base SNom)" annotation(
+    Placement(visible = true, transformation(origin = {110, -19}, extent = {{-17, -17}, {17, 17}}, rotation = 0), iconTransformation(origin =  {102, -20}, extent = {{-8, -8}, {8, 8}}, rotation = 0)));
+  ComplexBlocks.Interfaces.ComplexOutput uPu (re(start = u0Pu.re), im(start=u0Pu.im)) "complex inverter terminal voltage" annotation(
+    Placement(visible = true, transformation(origin = {110, 59}, extent = {{-17, -17}, {17, 17}}, rotation = 0), iconTransformation(origin =  {102, 50}, extent = {{-8, -8}, {8, 8}}, rotation = 0)));
+  // Internal variables:
+  Types.Angle UPhase (start = UPhase0) "Rotor angle: angle between machine rotor frame and port phasor frame";  
+
+protected 
+  
+  parameter Types.VoltageModulePu U0Pu  "Start value of voltage amplitude at injector terminal in p.u (base UNom)";
+  parameter Types.Angle UPhase0  "Start value of voltage angle at injector terminal (in rad)";
+  parameter Types.ActivePowerPu P0Pu  "Start value of active power in p.u (base SnRef) (receptor convention)";
+  parameter Types.ReactivePowerPu Q0Pu  "Start value of reactive power in p.u (base SnRef) (receptor convention)";
+    
+  parameter Types.ComplexVoltagePu u0Pu  "Start value of complex voltage at injector terminal in p.u (base UNom)";
+  parameter Types.ComplexApparentPowerPu s0Pu  "Start value of apparent power at injector terminal in p.u (base SnRef) (receptor convention)";
+  parameter Types.ComplexCurrentPu i0Pu  "Start value of complex current at injector terminal in p.u (base UNom, SnRef) (receptor convention)";
+    
+  parameter Types.PerUnit Id0Pu "Start value of id in p.u (base SNom)";
+  parameter Types.PerUnit Iq0Pu "Start value of iq in p.u (base SNom)";
+      
+equation
+
+  UPhase = ComplexMath.arg(terminal.V);
+  UPu = ComplexMath.'abs'(terminal.V);
+  uPu = terminal.V;
```

Comment:
At first sight, it is not self-explanatory why we have to define uPu, maybe add a comment in its declaration on why we do so instead of using directly terminal.V (terminal connector is for network connections only ?)

Reply:
done